### PR TITLE
chore(rln-relay): barrel imports

### PR DESCRIPTION
**Pull request by rymnc**
This PR introduces barrel imports for waku-rln-relay, resolving part of #1376.

Out of scope for this PR -
- Resolving cyclic dependencies by updating rln-relay imports
- Mounting procedure update in waku_node

### CODE REVIEW

File: waku/v2/protocol/waku_rln_relay/utils.nim [hunk 1]
```diff
@@ -14,11 +14,13 @@ import
...
+  stew/[byteutils, arrayops, endians2]
+import
   rln, 
-  waku_rln_relay_constants,
-  waku_rln_relay_types,
-  waku_rln_relay_metrics,
+  constants,
+  types,
+  protocol_metrics
+import
```
Comment (LNSD) [inline waku/v2/protocol/waku_rln_relay/utils.nim:23]: [_use explicit import paths (e.g. ./ for modules in same path_)
](https://hackmd.io/5oIVOuQcSM2WKrnDLUPcpg?view#:~:text=use%20explicit%20import%20paths%20(e.g.%20./%20for%20modules%20in%20same%20path))

File: waku/v2/protocol/waku_rln_relay/protocol_metrics.nim [hunk 2]
```diff
@@ -8,7 +8,7 @@ import
   chronos,
   metrics,
   metrics/chronos_httpserver,
-  waku_rln_relay_constants,
```
Comment (LNSD) [inline waku/v2/protocol/waku_rln_relay/protocol_metrics.nim:11]: What are these constants related to? Static protocol? In that case, the module name should've been something likie `protocol_static_constants`

File: waku/v2/node/waku_node.nim [hunk 3]
```diff
@@ -44,7 +44,7 @@ import
 
 when defined(rln):
   import
-    ../protocol/waku_rln_relay/waku_rln_relay_types
+    ../protocol/waku_rln_relay/types
```
Comment (LNSD) [inline waku/v2/node/waku_node.nim:47]: Here we should also import the barrel import module

File: waku/v2/node/waku_metrics.nim [hunk 4]
```diff
@@ -17,7 +17,7 @@ import
   ./waku_node
 
 when defined(rln):
-  import ../protocol/waku_rln_relay/waku_rln_relay_metrics
+  import ../protocol/waku_rln_relay/protocol_metrics as rln_metrics
```
Comment (LNSD) [inline waku/v2/node/waku_metrics.nim:20]: Here we should also import the barrel import module

File: apps/wakunode2/wakunode2.nim [hunk 5]
```diff
@@ -47,8 +47,7 @@ import
 
 when defined(rln):
   import
-    ../../waku/v2/protocol/waku_rln_relay/waku_rln_relay_types,
-    ../../waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils
+    ../../waku/v2/protocol/waku_rln_relay/utils
```
Comment (LNSD) [inline apps/wakunode2/wakunode2.nim:50]: Here we should also import the barrel import module

File: apps/chat2/chat2.nim [hunk 6]
```diff
@@ -38,7 +38,7 @@ when defined(rln):
   import
     libp2p/protocols/pubsub/rpc/messages,
     libp2p/protocols/pubsub/pubsub,
-    ../../waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils
+    ../../waku/v2/protocol/waku_rln_relay/utils
```
Comment (LNSD) [inline apps/chat2/chat2.nim:41]: Here we should also import the barrel import module

File: waku/v2/protocol/waku_rln_relay/utils.nim (hunk 1 above)
Comment (rymnc) [reply waku/v2/protocol/waku_rln_relay/utils.nim]: Addressed in e9d15c53

File: waku/v2/protocol/waku_rln_relay/rln.nim [hunk 7]
```diff
@@ -7,7 +7,7 @@ else:
 
 import
   os,
-  waku_rln_relay_types
```
[the remark this answers is not in the document]
Comment (rymnc) [reply waku/v2/protocol/waku_rln_relay/rln.nim:10]: Addressed in e9d15c53

File: waku/v2/protocol/waku_rln_relay/protocol_metrics.nim (hunk 2 above)
Comment (rymnc) [reply waku/v2/protocol/waku_rln_relay/protocol_metrics.nim:11]: No, these are constants generally related to rln-relay, like maximum epoch between messages, etc. This will be refactored soon.

File: waku/v2/protocol/waku_message.nim [hunk 8]
```diff
@@ -20,7 +20,7 @@ import
 
 when defined(rln):
   import
-    ./waku_rln_relay/waku_rln_relay_types
+    ./waku_rln_relay/types
```
[the remark this answers is not in the document]
Comment (rymnc) [reply waku/v2/protocol/waku_message.nim]: Addressed in https://github.com/status-im/nwaku/commit/e9d15c538c906cba1b2a87d5fa31d301e991e2df. The reason I didn't do this initially was that the next PR could have included updating imports + resolving cyclic dependencies. Sorry :)

File: waku/v2/node/waku_node.nim (hunk 3 above)
Comment (rymnc) [reply waku/v2/node/waku_node.nim]: Addressed in https://github.com/status-im/nwaku/commit/e9d15c538c906cba1b2a87d5fa31d301e991e2df

File: apps/chat2/chat2.nim (hunk 6 above)
Comment (rymnc) [reply apps/chat2/chat2.nim]: Addressed in https://github.com/status-im/nwaku/commit/e9d15c538c906cba1b2a87d5fa31d301e991e2df

**Review by LNSD — APPROVED** [review]
LGTM ✅

File: waku/v2/node/waku_node.nim (hunk 3 above)
Comment (rymnc) [reply waku/v2/node/waku_node.nim]: You want this fix in this PR?

Comment (LNSD) [reply waku/v2/node/waku_node.nim]: I should have been more explicit regarding that 🙏🏼 

No, out of the scope of this PR. Here you have a draft PR for that. Would you mind taking it over? 😄

**Comment by LNSD** [comment]
> This is a great step for the RLN refactoring!
> 
> I see the CI workflows are failing, but it seems for something related to the nimbus-build-system. I wonder if such error could be related to what observed in #1385 @LNSD.

I see [here](https://github.com/status-im/nwaku/actions/runs/3496209617/jobs/5853856400#step:7:125) that a module was renamed, but it was not updated in a test file.

```log
/Users/runner/work/nwaku/nwaku/tests/v2/test_waku_rln_relay_onchain.nim(11, 40) Error: cannot open file: ../../waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils
```

**Comment by LNSD** [comment]
Merging as @rymnc is OOO and the work here is finished
